code/utils.py

```diff
@@ -280,32 +283,55 @@ def extract_emb_desgn(emailids, embs):
     return np.array(X), np.array(y)
 
 
-def k_fold_cross_validation(email_ids, embs):
-    # extract the data
-    X, y = extract_emb_desgn(email_ids, embs)
-    # split the data into k-folds
-    kf = KFold(n_splits=23, shuffle=True)
-    # run k-fold cross validation
-    cor = 0
-    y_t = np.array([])
-    y_p = np.array([])
-    for train_index, test_index in kf.split(X):
-        X_train = X[train_index]
-        y_train = y[train_index]
-        X_test = X[test_index]
-        y_test = y[test_index]
-        classifier = SVC()
-        classifier.fit(X_train, y_train)
-        y_pred = classifier.predict(X_test)
-        cor = cor + np.sum(y_pred == y_test)
-        y_t = np.append(y_t, y_test)
-        y_p = np.append(y_p, y_pred)
-    print (cor * 1.0) / len(y)
-
-    #plot the confusion matrix
-    confusion_matrix(y_t, y_p)
+def get_dominance_data(email_ids, embs):
+    """
+    :param email_ids: Email Ids of Enron people corresponding to their embedding
+    :param embs: Trained embeddings for people
+    :return X: concatenated pair of embeddings emb1,emb2
+    :return y: -1, 0 or 1 where -1 means desgn(emb1)<desgn(emb1), 0 when desgn(emb1)=desgn(emb1) &
+     +1 when desgn(emb1)>desgn(emb1)
+    """
+    desgn_order = {'Employee': 0,
+                   'Trader': 0,
+                   'Manager': 1,
+                   'In House Lawyer': 1,
+                   'Managing Director': 2,
+                   'Director': 3,
+                   'Vice President': 4,
+                   'President': 5,
+                   'CEO': 6}
+    user_desgn = load_user_designations(cat='None')
+    dom_embs = []
+    dom_y = []
+    for i in range(len(email_ids)):
+        for j in range(i+1, len(email_ids)):
+            if email_ids[i] not in user_desgn or email_ids[j] not in user_desgn or i == j:
+                continue
+            user_dg_i = user_desgn[email_ids[i]]
+            user_dg_j = user_desgn[email_ids[j]]
+            p = np.random.rand()
```

Comment:
We should be able to vectorize this and `get_dominance_relation`
But since it is just around 100*100, it might not be worth prioritizing it.